File: OTCDerivativesCalculatorModule/Project_Cpp/lib_static/FpmlSerialized/GenClass/fpml-msg-5-4/RequestMessage.cpp
// RequestMessage.cpp 
#include "RequestMessage.hpp"
#ifdef ConsolePrint
    #include <iostream>
#endif 

namespace FpmlSerialized {

RequestMessage::RequestMessage(TiXmlNode* xmlNode)
: Message(xmlNode)
{
    #ifdef ConsolePrint
        std::string initialtap_ = FileManager::instance().tap_;
        FileManager::instance().tap_.append("   ");
    #endif 
   //headerNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* headerNode = xmlNode->FirstChildElement("header");

   if(headerNode){headerIsNull_ = false;}
   else{headerIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- headerNode , address : " << headerNode << std::endl;
   #endif
   if(headerNode)
   {
      if(headerNode->Attribute("href") || headerNode->Attribute("id"))
      {
          if(headerNode->Attribute("id"))
          {
             headerIDRef_ = headerNode->Attribute("id");
             header_ = boost::shared_ptr<RequestMessageHeader>(new RequestMessageHeader(headerNode));
             header_->setID(headerIDRef_);
             IDManager::instance().setID(headerIDRef_,header_);
          }
          else if(headerNode->Attribute("href")) { headerIDRef_ = headerNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { header_ = boost::shared_ptr<RequestMessageHeader>(new RequestMessageHeader(headerNode));}
   }

   //validationNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* validationNode = xmlNode->FirstChildElement("validation");

   if(validationNode){validationIsNull_ = false;}
   else{validationIsNull_ = true;}

   if(validationNode)
   {
      for(validationNode; validationNode; validationNode = validationNode->NextSiblingElement("validation")){
          #ifdef ConsolePrint
              FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- validationNode , address : " << validationNode << std::endl;
          #endif
          if(validationNode->Attribute("href") || validationNode->Attribute("id"))
          {
              if(validationNode->Attribute("id"))
              {
                  validationIDRef_ = validationNode->Attribute("id");
                  validation_.push_back(boost::shared_ptr<Validation>(new Validation(validationNode)));
                  validation_.back()->setID(validationIDRef_);
                  IDManager::instance().setID(validationIDRef_, validation_.back());
              }
              else if(validationNode->Attribute("href")) { validationIDRef_ = validationNode->Attribute("href");}
              else { QL_FAIL("id or href error"); }
          }
          else { validation_.push_back(boost::shared_ptr<Validation>(new Validation(validationNode)));}
      }
   }
   else {
       #ifdef ConsolePrint
           FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- validationNode , address : " << validationNode << std::endl;
       #endif
   }

    #ifdef ConsolePrint
        FileManager::instance().tap_ = initialtap_;
    #endif 
}
boost::shared_ptr<RequestMessageHeader> RequestMessage::getHeader()
{
   if(!this->headerIsNull_){
        if(headerIDRef_ != ""){
             return boost::shared_static_cast<RequestMessageHeader>(IDManager::instance().getID(headerIDRef_));
        }else{
             return this->header_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<RequestMessageHeader>();
   }
}
std::vector<boost::shared_ptr<Validation>> RequestMessage::getValidation()
{
   if(!this->validationIsNull_){
        if(validationIDRef_ != ""){
             return this->validation_;
        }else{
             return this->validation_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return std::vector<boost::shared_ptr<Validation>>();
   }
}
}

File: OTCDerivativesCalculatorModule/Project_CSharp/XmlBuilder/XmlSerializedClass/fpml-msg-5-4/RequestMessage.cs
using System;
using System.Text;
using System.Collections.Generic;
using System.Xml;

namespace FpmlSerializedCSharp
{
    public class RequestMessage : Message
    {
        public RequestMessage(XmlNode xmlNode)
        : base(xmlNode)
        {
            XmlNode headerNode = xmlNode.SelectSingleNode("header");
            
            if (headerNode != null)
            {
                if (headerNode.Attributes["href"] != null || headerNode.Attributes["id"] != null) 
                {
                    if (headerNode.Attributes["id"] != null) 
                    {
                        headerIDRef_ = headerNode.Attributes["id"].Value;
                        RequestMessageHeader ob = new RequestMessageHeader(headerNode);
                        IDManager.SetID(headerIDRef_, ob);
                    }
                    else if (headerNode.Attributes["href"] != null)
                    {
                        headerIDRef_ = headerNode.Attributes["href"].Value;
                    }
                    else
                    {
                        header_ = new RequestMessageHeader(headerNode);
                    }
                }
                else
                {
                    header_ = new RequestMessageHeader(headerNode);
                }
            }
            
        
            XmlNodeList validationNodeList = xmlNode.SelectNodes("validation");
            
            if (validationNodeList != null)
            {
                this.validation_ = new List<Validation>();
                foreach (XmlNode item in validationNodeList)
                {
                    if (item.Attributes["href"] != null || item.Attributes["id"] != null) 
                    {
                        if (item.Attributes["id"] != null) 
                        {
                            validationIDRef_ = item.Attributes["id"].Value;
                            validation_.Add(new Validation(item));
                            IDManager.SetID(validationIDRef_, validation_[validation_.Count - 1 ]);
                        }
                        else if (item.Attributes["href"] != null)
                        {
                            validationIDRef_ = item.Attributes["href"].Value;
                        }
                        else
                        {
                        validation_.Add(new Validation(item));
                        }
                    }
                    else
                    {
                        validation_.Add(new Validation(item));
                    }
                }
            }
            
        
        }
        
    
        #region Header_
        private RequestMessageHeader header_;
        public RequestMessageHeader Header_
        {
            get
            {
                if (this.header_ != null)
                {
                    return this.header_; 
                }
                else if (this.headerIDRef_ != null)
                {
                    header_ = IDManager.getID(headerIDRef_) as RequestMessageHeader;
                    return this.header_; 
                }
                else
                {
                      return this.header_; 
                }
            }
            set
            {
                if (this.header_ != value)
                {
                    this.header_ = value;
                }
            }
        }
        #endregion
        
        public string headerIDRef_ { get; set; }
        #region Validation_
        private List<Validation> validation_;
        public List<Validation> Validation_
        {
            get
            {
                if (this.validation_ != null)
                {
                    return this.validation_; 
                }
                else if (this.validationIDRef_ != null)
                {
                    return this.validation_; 
                }
                else
                {
                      return this.validation_; 
                }
            }
            set
            {
                if (this.validation_ != value)
                {
                    this.validation_ = value;
                }
            }
        }
        #endregion
        
        public string validationIDRef_ { get; set; }
        
    
        
    
    }
    
}

